### ISSUE TITLE
App not working when ran on iPhone simulator on iOS 14

### ISSUE THREAD

**Issue by El-Vegas** [beeware/briefcase#508]
I made a simple app (using python yahoo_fin module to show one stock price) following tutorial. Everything (almost everything) worked perfectly (including macOS app) until I tried to run iOS app in xCode. The app doesn't show any errors, it is just that it exits itself right after the app screen launches. Could it be that iOS version is too new (iOS 14) that maybe not everything is updated to this version? It worked with macOS Catalina version 10.15.7.

**Comment by freakboy3742** [comment]
I don't know anything about `yahoo_fin`, but I'm guessing you've hit #471. Looking at the PyPI definition for yahoo_fin, I'm guessing the problem is one of the dependencies of `yahoo_fin`, since `yahoo_fin` itself appears to be a pure python package.

**Comment by El-Vegas** [comment]
I see. Thanks

**Comment by El-Vegas** [comment]
I actually get the same result when I run just a simple Hello World app as in tutorial. 

This is the last line that I get: com.example.helloworld: 6822

**Comment by freakboy3742** [comment]
Is that with a *clean* build of the app? If you were experimenting adding `yahoo_fin`, it's possible that you've still got old code in the build that is being delivered to the phone.

To be completely sure - delete the iOS folder that briefcase created, and re-run `briefcase create` then `briefcase run`. (You could also use `briefcase update` then `briefcase build` then `briefcase run`, starting from clean is the easiest way to eliminate any possibility of error)

**Comment by El-Vegas** [comment]
This was a new build, I still kept the yahoo_fin app in the folder beeware-tutorial and I added another app/folder (helloworld) to beeware-tutorial folder and this second one didn't work, so I retried one more time, I deleted helloworld and remade it again with the only code the title on top of it (Hello World). Same result at the end.

**Comment by The-Judge** [comment]
This happens with the "bare" Hello World - App from the Beeware - Tutorial even, with a state from "[Tutorial 2 - Making it interesting](https://docs.beeware.org/en/latest/tutorial/tutorial-2.html#tutorial-2-making-it-interesting)". No external dependencies other than used in the Tutorial used.

It seems to have nothing to do with iOS 14; for me, it isn't working with iOS 13 and 12 either. Here's what I did:

1. Starting Xcode
2. Clicking "Xcode > Preferences" in menu
3. Clicking "Components" tab
4. Clicking the down-arrow symbol left to any of the iOS versions listed below
5. After waiting for the 3 - 4 GB per version download to finish, starting the app with a specific iOS emulator device like this: `briefcase run iOS -d "iPhone 11::iOS 13.x"`

I tried with 12.4 through to 14.1 this way with the following results:

| iOS version   | Result   |
| ------------- | ------------- |
| 12.4 | crashing immediately after showing splash screen |
| 13.3 | crashing immediately after showing splash screen |
| 13.4 | crashing immediately after showing splash screen |
| 13.5 | crashing immediately after showing splash screen |
| 13.7 | crashing immediately after showing splash screen |
| 14.1 | crashing immediately after showing splash screen |

### My versions:
* OS: MacOSX 11.0.1
* Python: 3.8.6

```
$ pip freeze
arrow==0.17.0
binaryornot==0.4.4
briefcase==0.3.3
certifi==2020.11.8
chardet==3.0.4
click==7.1.2
cookiecutter==1.7.2
dmgbuild==1.4.2
ds-store==1.3.0
gitdb==4.0.5
GitPython==3.1.11
idna==2.10
Jinja2==2.11.2
jinja2-time==0.2.0
mac-alias==2.1.0
MarkupSafe==1.1.1
poyo==0.5.0
python-dateutil==2.8.1
python-slugify==4.0.1
requests==2.25.0
rubicon-objc==0.4.0
six==1.15.0
smmap==3.0.4
text-unidecode==1.3
toga-cocoa==0.3.0.dev25
toga-core==0.3.0.dev25
toml==0.10.2
travertino==0.1.3
urllib3==1.26.2
```

Please find a demo of this behavior at https://www.youtube.com/watch?v=E4OrreeNpIA

**Comment by synergit** [comment]
Hello All, do we find the root cause? My gut feeling is that the root cause is the new iOS version 14.3. So I am wondering if we have workaround, like how to target older version? Not sure whether I tried hard enough, but with new macOS Big Sur v11.1, XCode simulator always targets iOS 14.3. 

Any help is appreciated!

**Comment by freakboy3742** [comment]
@The-Judge Apologies for the very delayed response. Based on the log you've provided, the app is failing when it tries to execute:

    xcrun simctl install 5E1F9452-B354-4F38-BCA5-508A6227CEC1 <app path>

where `<app path>` is the file system location of the app. My best guess is that the binary path is either failing to evaluate, or has a character that can't be converted. Can I get you to provide:
1. The full path you're using for your app
2. The formal name and app name for your app
3. If you're able: set a breakpoint in the code at the point that is raising the exception, and provide the value of `arg` and `args` when the exception is raised.

@synergit Your `register() takes from 2 to 3 positional arguments` issue is slightly different - around the date of your comment, we had an issue with registration of iOS web browser support; that was fixed in the 3.8-b4 support package.

**Comment by freakboy3742** [comment]
Closing due to lack of response from the original reporter.
